vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.jsp

```diff
@@ -48,38 +48,58 @@ function onDoubleClick(e){
         // remove highlighting if present
         $(this).removeClass("hova");
     } else {
-        $("#addImgAlert").show();
+    	
+    	// storing image ID selected by the user to replace, onDoubleClick
+    	var imgID = $(e.target).attr('id'); 
+    	$("#uploadImage").data('value', imgID) // sets image ID value
+    	$("#addImgAlert").show();
+       	
     }
 }
     
 function uploadImage() {
-    var file = document.getElementById('file').files[0];
+   	
+	var imgID = $("#uploadImage").data('value') // gets image ID
+    var file = $('#file')[0].files[0]
     var reader  = new FileReader();
     var formData = new FormData();
     formData.append('file', file);
     formData.append('contentOrder', contentCount);
-    if ($(".open")[0]){
-        var imageBlockId = $('.open img').attr('id')
-        formData.append('imageBlockId',imageBlockId);
-        var url = "<c:url value="/staff/module/${module.id}/slide/${slide.id}/image/" />" + imageBlockId + "?${_csrf.parameterName}=${_csrf.token}";
-        reader.onload = function () {
-            imageblock = createImageBlock(reader);
-            $("#" + imageBlockId).replaceWith(imageblock);
+    var imageblock = "";
```

Comment:
unless ```imageblock``` is used outside the if/else statement, it should be declared where needed

Reply:
Okay, I will change that.

Reply:
imageblock - is being outside if/else statement, hence I declared it outside.